app/controllers/carts_controller.rb

```diff
@@ -15,11 +15,13 @@ def create
 
   def update
     item_id = params[:id]
-    condition = params[:condition]
-    @cart.item_quantity(item_id, condition)
-    if @cart.contents[item_id] == 0
-      @cart.delete_item(item_id)
-      flash[:successfully_removed] = "Successfully removed <a href=#{item_path(item_id)}>#{Item.find(item_id).title}</a> from your cart."
```

Comment:
I like this refactor, makes this method a little fatter, but the logic looks good.

Reply:
@samuelssnider Thanks Sam!